### PR TITLE
[3.x] Add missing argument to `AnimationNodeBlendTreeEditor::_delete_nodes_request`

### LINKED ISSUE #65615: Deleting AnimationTree node with Delete key doesn't delete, shows error.
### Godot version

3.5 stable

### System information

Debian 11, AMD Ryzen™ 5 3400G with Radeon™ RX Vega 11 Graphics

### Issue description

When pressing Delete key in a AnimationNodeBlendTree node, it stays in the grid (it isn't deleted) and Godot shows this error:

    • core/object.cpp:1242 - Error calling method from signal 'delete_nodes_request': 'AnimationNodeBlendTreeEditor::_delete_nodes_request': Method expected 0 arguments, but called with 1..

In Godot 4.0 Alpha 12, node is correctly deleted.

### Steps to reproduce

- Create AnimationPlayer node
- Create AnimationTree node
- In AnimationTree Inspector, create a AnimationNodeBlendTree
- Link AnimationTree with AnimationPlayer
- Go to AnimationTree
- Create any node (e.g Animation)
- Click on node
- Press Delete key.
- Look at Output messages.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by asalazar516**
Fixes #65615 

From @kleonc comment, back ported changes from master on delete_nodes_request to 3.x branch.

This is my first pull request. Please let me know if I need to do any additional changes.

### CODE REVIEW

File: editor/plugins/animation_blend_tree_editor_plugin.cpp [hunk 1]
```diff
@@ -456,17 +456,22 @@ void AnimationNodeBlendTreeEditor::_delete_request(const String &p_which) {
...
+				if (gn->is_selected() && gn->is_close_button_visible()) {
+					to_erase.push_back(gn->get_name());
+				}
 			}
 		}
-	}
+	} else {
+		for (int i = 0; i < p_nodes.size(); i++) {
+			to_erase.push_back(p_nodes[i]);
+		}
```
Comment (kleonc) [inline editor/plugins/animation_blend_tree_editor_plugin.cpp:474]:
```suggestion
		}
	}
```
The closing parenthesis in the `else` block is missing.

**Comment by kleonc** [comment]
> This is my first pull request.

Thanks for contributing!

>Please let me know if I need to do any additional changes.

Seems like you missed a single closing parenthesis, I've suggested a change. For how-to update this PR refer to the docs: [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#making-changes) (please keep the changes in a single commit). Besides that the code looks good.

> Change made for issue #65615

You can edit that comment and write something like _"Fixes #65615"_ instead to make this PR be automatically linked to the given issue so the issue would be auto-closed when this PR is merged. See: [Linking a pull request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

---

Also regarding the PR/commit title: _"Backport AnimationNodeBlendTreeEditor to 3.x"_ could be a little misleading, when I read it it makes me think `AnimationNodeBlendTreeEditor` were not yet in 3.x which is not true. So I'd suggest rewording PR/commit titles to e.g. something like: _"[3.x] Add missing argument to `AnimationNodeBlendTreeEditor::_delete_nodes_request`"_ which better describes what's actually done in here.

**Comment by asalazar516** [comment]
Thanks, @kleonc for the help.  Updated changes. I will also visit the chat, next time, for any other questions.

**Comment by kleonc** [comment]
As you can see there are 2 commits now, as I've mentioned a single commit is preferred. You'd need to `rebase`, for how-to see: [the interactive rebase](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase)  (you'd need to: `fixup` the second commit, `reword` the first commit as suggested in the previous comment (you've renamed PR only, not the commit), and force push the changes so it will update in here).

Also you haven't done that:


> > Change made for issue #65615
> 
> You can edit that comment and write something like _"Fixes #65615"_ instead to make this PR be automatically linked to the given issue so the issue would be auto-closed when this PR is merged. See: [Linking a pull request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

(I could edit it myself but I assume you want to learn it yourself)

**Comment by asalazar516** [comment]
@kleonc Sorry for the trouble. This is the first time I ever used rebase, before.

- Updated the comment
- Used rebase to _fixup_ the second commit and _reword_ the first commit.

**Comment by asalazar516** [comment]
@kleonc Thank you so much for the help! And I see I was rebasing on my own branch and using _fixup_ oppose to drop. The branch now has just the one commit. Thank you also for linking the PR.

Very much appreciate the help. Got a much better understanding with PR process, next time I try to help.

File: editor/plugins/animation_blend_tree_editor_plugin.cpp [hunk 2]
```diff
@@ -456,16 +456,22 @@ void AnimationNodeBlendTreeEditor::_delete_request(const String &p_which) {
 	undo_redo->commit_action();
 }
 
-void AnimationNodeBlendTreeEditor::_delete_nodes_request() {
+void AnimationNodeBlendTreeEditor::_delete_nodes_request(const TypedArray<StringName> &p_nodes) {
```
Comment (kleonc) [inline editor/plugins/animation_blend_tree_editor_plugin.cpp:459]: In 3.x there's no typed arrays.
```suggestion
void AnimationNodeBlendTreeEditor::_delete_nodes_request(const Array &p_nodes) {
```
also you need to add the same parameter to the declaration of this method:
https://github.com/godotengine/godot/blob/01574a566ff1f1a7270bd986b226c1bd351f894e/editor/plugins/animation_blend_tree_editor_plugin.h#L106

Just check what was done in #61145. Also... in theory before pushing changes to the PR you should compile and test them on your end to ensure it actually works. :wink:

**Review by kleonc — CHANGES_REQUESTED** [review]

File: editor/plugins/animation_blend_tree_editor_plugin.cpp (hunk 2 above)
Comment (kleonc) [reply editor/plugins/animation_blend_tree_editor_plugin.cpp:459]: And to apply a suggestion like this you can do it like before (click button like "commit suggestion" on GitHub, end up with 2 commits, then fixup+ force push), or just manually ammend the changes to your local commit and force push.

Comment (asalazar516) [reply editor/plugins/animation_blend_tree_editor_plugin.cpp]: Sounds good! My bad on the compiling. I just amend the changes. Tested the build, and was able to compile and verified the bug was resolved.

Comment (kleonc) [reply editor/plugins/animation_blend_tree_editor_plugin.cpp]: Great! So just a matter of passing the tests and it should be good to merge. :slightly_smiling_face:

**Review by kleonc — APPROVED** [review]

**Comment by kleonc** [comment]
@akien-mga Should be good to merge.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by timothyqiu** [comment]
Cherry-picked for 3.5.2
